tests/unit/frameworks/test_any_agent.py

```diff
@@ -1,18 +1,147 @@
+        ]
+    }
+
+    yield {
+        "choices": [
+            {
+                "delta": {"content": "Pennsylvania is "},
+                "index": 0,
+                "finish_reason": None,
+            }
+        ]
+    }
+
+    # Final chunk with finish reason
+    yield {
+        "choices": [
+            {"delta": {"content": "Harrisburg."}, "index": 0, "finish_reason": "stop"}
+        ]
+    }
+
+
+class TestAgentCreation:
+    def test_create_any_with_framework(self, agent_framework: AgentFramework) -> None:
+        agent = AnyAgent.create(agent_framework, AgentConfig(model_id="gpt-4o"))
+        assert agent
+
+    def test_create_any_with_valid_string(
+        self, agent_framework: AgentFramework
+    ) -> None:
+        agent = AnyAgent.create(agent_framework.name, AgentConfig(model_id="gpt-4o"))
+        assert agent
+
+    def test_create_any_with_invalid_string(self) -> None:
+        with pytest.raises(ValueError, match="Unsupported agent framework"):
+            AnyAgent.create("non-existing", AgentConfig(model_id="gpt-4o"))
+
+
+# Tests for model arguments
+class TestModelArguments:
+    def test_model_args(
+        self, agent_framework: AgentFramework, mock_litellm_response: Any
+    ) -> None:
+        if agent_framework == AgentFramework.LLAMA_INDEX:
+            pytest.skip("LlamaIndex agent uses a litellm streaming syntax")
+
+        agent = create_agent_with_model_args(agent_framework)
+
+        # Patch the appropriate litellm import path for this framework
+        import_path = LITELLM_IMPORT_PATHS[agent_framework]
+        with patch(import_path) as mock_litellm:
+            # Configure the mock
+            mock_litellm.return_value = mock_litellm_response
+
+            # Run the agent
+            result = agent.run(TEST_QUERY)
+
+            # Verify results
+            assert EXPECTED_OUTPUT == result.final_output
+            assert mock_litellm.call_args.kwargs["temperature"] == TEST_TEMPERATURE
+            assert mock_litellm.call_args.kwargs["frequency_penalty"] == TEST_PENALTY
+            assert mock_litellm.call_count > 0
 
-def test_create_any_with_framework(agent_framework: AgentFramework) -> None:
-    agent = AnyAgent.create(agent_framework, AgentConfig(model_id="gpt-4o"))
-    assert agent
+    def test_model_args_streaming(self, agent_framework: AgentFramework) -> None:
+        if agent_framework != AgentFramework.LLAMA_INDEX:
+            pytest.skip("This test is only for LlamaIndex framework")
 
+        agent = create_agent_with_model_args(agent_framework)
 
-def test_create_any_with_valid_string(agent_framework: AgentFramework) -> None:
-    agent = AnyAgent.create(agent_framework.name, AgentConfig(model_id="gpt-4o"))
-    assert agent
+        # Patch the appropriate litellm import path for LlamaIndex
+        import_path = LITELLM_IMPORT_PATHS[agent_framework]
+        with patch(import_path) as mock_litellm:
```

Comment:
Again, it might be shortened to this?
```suggestion
        with patch(import_path, side_effect=mock_streaming_response) as mock_litellm:
```